src/lib/aws-sdk/cloudwatch.ts

```diff
@@ -0,0 +1,50 @@
+import * as AWS from 'aws-sdk';
+import { validateCredentials } from './credentials';
+import { debug } from '../logger';
+import { match } from '../utils';
+
+export async function getCloudWatchMetricAlarms(
```

Comment:
Why are both of these a copy-paste? As far as I see only the `res.CompositeAlarms`/`res.MetricAlarms` is the only difference. Why not modularize this code not to duplicate the whole thing?

Reply:
The amount of code that would be required to modularize that would be greater than just copying

Reply:
Amount of code required isn't the only indicator that should be taken into the consideration when providing a solution, otherwise we would not write tests at all not to increase the codebase. Long term maintainability and scalability of our code is far more important and outweighs the downside of modularization you mentioned.

Reply:
Modularized the code, does it look good now?

Reply:
Using union types would allow you to retain an initial return structure, but the way you did it is also good. :+1: